lib/search/query_components/aggregates.rb

```diff
@@ -0,0 +1,70 @@
+module QueryComponents
+  class Aggregates < BaseComponent
+    def payload
+      search_params.aggregates.reduce({}) do |result, (field_name, options)|
+        result[field_name] = aggregates_hash_for_aggregate(field_name, options)
+        result["#{field_name}_with_missing_value"] = aggregates_hash_for_null_values(field_name, options)
+        result
+      end
+    end
+
+  private
+
+    def aggregates_hash_for_aggregate(field_name, options)
+      with_filters(
+        {
+          terms: {
+            field: field_name,
+            order: { _count: "desc" },
+            # We want all the aggregate values so we can return an accurate count of
+            # the number of options.
+            size: 100000,
```

Comment:
I'm not really sure, I have left this as is for the moment add will add to the backlog for review/refactoring